src/tree_data.c

```diff
@@ -2796,25 +2879,33 @@ lyd_compare_single(const struct lyd_node *node1, const struct lyd_node *node2, u
         }
     }
 
-    if ((LYD_CTX(node1) != LYD_CTX(node2)) || (node1->schema != node2->schema)) {
+    if (node1->hash != node2->hash) {
         return LY_ENOT;
     }
+    /* equal hashes do not mean equal nodes, they can be just in collision (or both be 0) so the nodes must be checked explicitly */
```

Comment:
I know you did not change this, but I do not think the hashes can be `0` so I would remove that.

Reply:
ok